bigtable-hbase-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -819,93 +832,174 @@ public boolean isTableAvailable(TableName tableName, byte[][] splitKeys) throws
   @Override
   public void snapshot(String snapshotName, TableName tableName)
       throws IOException, SnapshotCreationException, IllegalArgumentException {
-    throw new UnsupportedOperationException("snapshot");  // TODO
+    waitForOperation(snapshotTable(snapshotName, tableName));
+  }
+
+  private Operation snapshotTable(String snapshotName, TableName tableName)
+      throws IOException {
+    return bigtableTableAdminClient.snapshotTable(SnapshotTableRequest.newBuilder()
```

Comment:
HBase snapshots are not cleaned up.  The current behavior would be consistent with HBase.  HBase should use our server-side default, what ever that is.